xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/editors/autosave.js

```diff
@@ -83,44 +83,40 @@ editors.AutoSave = Class.create({
   },
 
   /**
-   * The UI of the autosave feature is created and introduced at the beginning of the edit form. It comprises a checkbox
-   * for enabling / disabling the autosave and an input that allows to set the autosave frequency.
+   * The UI of the autosave feature is created and introduced towards the end of the edit form. 
+   * It contains a checkbox for toggling the autosave and an input that allows to set the autosave interval in minutes.
    */
   createUIElements : function() {
-    // Checkbox to enable/disable the autosave
+    // Toggle for the autosave feature.
     this.autosaveCheckbox = new Element('input', {
       type: "checkbox",
       checked: this.options.enabled,
       name: "doAutosave",
       id: "doAutosave"
     });
-    // Input for setting the autosave frequency
-    this.autosaveInput = new Element('input', {
+    // Input for setting the autosave interval
+    this.autosaveIntervalInput = new Element('input', {
       type: "text",
-      value: this.options.frequency,
+      value: this.options.interval,
       size: "2",
-      "class": "autosave-frequency"
+      "class": "autosave-interval"
     });
     // Labels
-    var autosaveLabel = new Element('label', {'class': 'autosave', 'for' : "doAutosave"});
+    let autosaveLabel = new Element('label', {'class': 'autosave'});
     autosaveLabel.appendChild(this.autosaveCheckbox);
-    autosaveLabel.appendChild(document.createTextNode(" $services.localization.render('core.edit.autosave')"));
-    var frequencyLabel = new Element('label', {'class': 'frequency'});
-    frequencyLabel.appendChild(document.createTextNode("$services.localization.render('core.edit.autosave.every') "));
-    frequencyLabel.appendChild(this.autosaveInput);
-    this.timeUnit = new Element('span');
-    this.setTimeUnit();
-    frequencyLabel.appendChild(document.createTextNode(" "));
-    frequencyLabel.appendChild(this.timeUnit);
+    autosaveLabel.appendChild(document.createTextNode("$escapetool.javascript($services.localization.render('core.edit.autosave'))"));
```

Comment:
It would be much simpler to use the [`append` method](https://developer.mozilla.org/en-US/docs/Web/API/Element/append) here (if this is a proper `Element` and not just something from Prototype, if not it might be a good reason to migrate away from Prototype).